packages/styled-components/src/types.ts

```diff
@@ -299,3 +299,8 @@ export type CSSProp = Interpolation<any>;
 export type NoInfer<T> = [T][T extends any ? 0 : never];
 
 export type Substitute<A extends object, B extends object> = FastOmit<A, keyof B> & B;
+
+export interface ThemeProps extends ExecutionContext {}
```

Comment:
ThemeProps was generic in @types/styled-components package: 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/styled-components/index.d.ts#L25C1-L28C1

Reply:
Yes, it is.

As mentioned in https://github.com/styled-components/styled-components/pull/4126#discussion_r1290796829, it needs to be placed on top of #4103. Or merge this first and then follow.
However, in some cases it should be possible to replace it by using ExecutionContext.